FreeRTOS/FreeRTOS-Kernel
# Update taskSELECT_HIGHEST_PRIORITY_TASK macro for SMP

Description
-----------
* Move the configUSE_PORT_OPTIMISED_TASK_SELECTION check to FreeRTOS.h
* SMP also use taskSELECT_HIGHEST_PRIORITY_TASK macro

Test Steps
-----------
N/A

Checklist:
----------



Related Issue
-----------
Address #750  comment in line [138](https://github.com/RichardBarry/FreeRTOS-Kernel/blob/814f078563eaf20ba9f59bd36cf48d85ae5e3185/tasks.c#L138C47-L138C124), [167](https://github.com/RichardBarry/FreeRTOS-Kernel/blob/814f078563eaf20ba9f59bd36cf48d85ae5e3185/tasks.c#L167)


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
